source/integrations/index.html

```diff
@@ -171,99 +318,241 @@
+        $filterElement.val(query);
+      }
+    }
+
+    // update all filter elements to match current filter obj
+    function updateFilterElements(queryFilter) {
+      console.log(`updateFilterElements():`);
+      let filterElements = [...document.querySelectorAll('.integration_filter select, input.search')];
+      let activeFilterElements = filterElements.filter(el => !(el.value === '#' || el.value === ''));
+
+      // clear filters if there's more than the current queryFilter
+      if (activeFilterElements.length > Object.keys(queryFilter).length) {
+        // filter out filter elements that don't match query object
+        let filtersToClear = activeFilterElements.filter(el => {
+          for (key in queryFilter) {
+             if (el.getAttribute('data-id') === key) return false;
+          }
+          return true;
+        });
+
+        // reset filter elements to default empty value
+        filtersToClear.forEach(filter => {
+          let queryId = filter.getAttribute('data-id');
+          if (queryId !== 'search') {
+            matchFilterToURL(queryId, '#');
+          } else {
+            matchFilterToURL(queryId, '');
+          }
+        });
+      } else {
+         // update new filters
+         if (Object.keys(queryFilter).length) {
+          for (key in queryFilter) {
+            matchFilterToURL(key, queryFilter[key]);
+          }
+        }
+      }
+    }
+
+    // filter button name cleanup. used inupdateFilterButtons()
+    function getButtonText(id) {
+      let text;
+      switch(id) {
+        case 'cat':
+          text = 'Category';
+          break;
+        case 'iot_class':
+          text = 'IoT';
+          break;
+        case 'quality_scale':
+          text = 'Quality';
+          break;
+        default:
+          text = id;
+      }
+      return text;
+    }
+
+    // update all filter buttons to match current obj filter
+    function updateFilterButtons(queryFilter) {
+      let filterButtonParent = document.querySelector('.component-search');
+      let currentFilterButtons = [...document.querySelectorAll('.component-search .active-filter')];
+      
+      // remove buttons if there's more than the current queryFilter
+      if (currentFilterButtons.length > Object.keys(queryFilter).length) {
+        // filter out buttons that dont match query obj
+        let buttonsToRemove = currentFilterButtons.filter(button => {
+          for (key in queryFilter) {
+             if (button.id === key) return false;
+          }
+          return true;
+        });
+        buttonsToRemove.forEach(button => button.remove());
+      } else {
+        //create a filter button if one doesn't already exists
+        for (key in queryFilter) {
+          if (document.querySelector(`.component-search .active-filter#${key}`) === null) {
+            let button = document.createElement('button');
+            button.innerHTML = `<span>${getButtonText(key)}<i class="icon-plus-sign"></i></span>`;
```

Comment:
this should include the filtered value as well, like `version: 2022.11.0` and `Category: featured`.

Reply:
updated the `updateFilterButtons` function. Now there is another check to test if the filter object values have changed, and it updates the button text accordingly. Also switched to use the data attribute over id